### PR TITLE
Fix issue of deleting partition when bad as3 schema.

### PR DESCRIPTION
**Pull request by zongzw**
**Description**:  

CIS should not delete existing config when configmap in bad format 

**Changes Proposed in PR**:

The issue happens because:
The coming configmap json(containing partitions to create) is fully parsed because of bad schema format.
Thus, by comparing existing configmap in CIS, the fails-to-parsed partition is considered as `deleteTenants`.

The fix can be: recording the parsing result, more concisely tell later logic to distinguish it's invalid or real-missing.

**Fixes**: resolves SR  1-8181523521 / connell-1 

## General Checklist
- [x] Updated Added functionality/ bug fix in release notes - TBD in release notes 
- [x] Smoke testing completed

### CODE REVIEW

File: pkg/agent/as3/as3Utils.go [hunk 1]
```diff
@@ -312,6 +320,13 @@ func getClass(obj interface{}) string {
 	return cl.(string)
 }
 
+func assertToBe(kind string, obj interface{}) bool {
+	if obj == nil {
+		return false
+	}
+	return (reflect.TypeOf(obj).Kind().String() == kind)
```
Comment (sravyap135) [inline pkg/agent/as3/as3Utils.go:327]: reflect package adds additional overhead. Can you revert these changes from this PR.

Comment (zongzw) [reply pkg/agent/as3/as3Utils.go:327]: The impact is 'ns'(nano seconds) level, but the deployment of declaration is 's' (seconds, in fact tens of seconds) level. They are totally not in the same page. 
A bit more, assertTobe(map)  can justify not only `map[string]interface{}` but also `map[string]string` or map[string]xxxx. It may have more usage abilities.

Comment (sravyap135) [reply pkg/agent/as3/as3Utils.go:327]: Benchmarking results - 

BenchmarkInterfaceTest-12 1000000000 **0.2527** ns/op 0 B/op 0 allocs/op

BenchmarkAssertToBe-12 482082412 **2.494** ns/op 0 B/op 0 allocs/op

-  reflect adds an additional overhead by a factor of 10X

**Comment by sravyap135** [comment]
Could you please update the PR with latest master so that I can initiate smoke test on this and we can target to merge it in 2.9.1

**Comment by sravyap135** [comment]
@zongzw I ran the smoke tests. Can you please rebase the PR, update release notes. I can trigger azure pipeline.

**Review by sravyap135 — APPROVED** [review]
